intro-to-javascript-testing.md

```diff
@@ -0,0 +1,142 @@
+* Install the `jshint` [command line interface](http://jshint.com/docs/cli/), and integrate it with the regular running of your test suite. Run `npm install -g jshint` to install JSHint globally. Then, in terminal, tell JSHint to lint your files. 
+
+The `jshint` cli has several options. Most simply, the `jshint` command accepts a specific file as a parameter:
+
+``` bash
+# Inspect this JS file!
+jshint my_unique_app/static/js/custom.js
+
+# Discouraging output...
+my_unique_app/static/js/custom.js: line 14, col 51, Missing semicolon.
+my_unique_app/static/js/custom.js: line 18, col 48, Missing semicolon.
+my_unique_app/static/js/custom.js: line 52, col 34, Missing semicolon.
+
+3 errors
+```
+
+Alternatively, JSHint can recursively discover all JS files in a directory and return a report: `jshint .` The results, in this case, may not be useful, since JSHint lints third-party libraries, too. For greater precision, add a configuration file in the root of your app, called `.jshintignore`. Here, tell JSHint to ignore, let's say, the directory where you store external libraries.
+
+```
+my_unique_app/static/js/lib/*
+```
+
+The JSHint cli allows for automated testing with Travis. Include in your `.travis.yml` file instructions for installing `jshint` and directive(s) for running the linter:
+
+```
+...
+
+install:
+- pip install --upgrade pip
+- pip install --upgrade -r requirements.txt
+- npm install -g jshint
+
+...
+
+script: 
+- pytest tests
+- jshint my_unique_app/static/js/custom.js
+
+...
+``` 
+
+## jasmine
+
+The right style and perfect syntax make for clean, readable, debuggable JavaScipt. Yet, testing JS functionality requires something distinct. For this, we recommend [Jasmine](https://jasmine.github.io/2.0/introduction.html), a testing framework for behavior-driven development. 
+
+### Python integration
+
+Jasmine [plays nicely with Django and Flask apps](https://jasmine.github.io/setup/python.html). In the virtualenv of your app, install Jasmine and initialize a project:
+
+```
+# Install
+pip install jasmine
+
+# Initialize a project
+jasmine-install
+```
+
+`jasmine-install` creates a `spec` repo, where all testing specs and the configuration file (i.e., `jasmine.yml`) live. Before writing and running tests, tell Jasmine where to find relevant files by specifying, most importantly, the `src_files` and `src_dir` attributes:
+
+```
+...
+
+src_files:
+  - js/*
+
+...
+
+src_dir: 'my_app/static'
+
+...
+```
+
+Then, in terminal, run:
+
+```bash
+jasmine
+```
+
+Visit `http://127.0.0.1:8888`, and view your test results.
```

Comment:
Do you have to view Jasmine output in a browser?

Reply:
I do not think so...but I'd like to open a Jasmine-specific issue about this and other questions.